### DISCUSSION TITLE
Extternal login wihtout local login leads to 404

### DISCUSSION THREAD

**Discussion by Justincale** [oqtane/oqtane.framework#4935] [Q&A]
HI, 

i have successfully implemented a custom external authentication provider and am able to login/out etc. I wish to turn off local login so that when someone clicks 'login' they are taken directly to the external provider but after turning off the 'allow login' user setting and clicking login, i receive a 404. Thanks

**Comment by sbwalker** [comment]
@Justincale are you using:

Version - 6.0.0
Render Mode - Static
Interactivity - Server

**Comment by sbwalker** [comment]
I cannot reproduce. I set up Entra authentication and disabled local Login. When I am logged out, if I hover over the Login button I can see that it points to /pages/external - which what invokes the external login flow:

![image](https://github.com/user-attachments/assets/a95fd147-c8f8-4184-8667-0c0d8bdd4ba1)

**Reply by Justincale** [reply]
Hi Shaun, I am running V6, Interactive Server, no pre-rendering. The issue seems to be that i am using my own custom theme with a custom login component. To reproduce.

Create a custom theme
Copy and use the Existing Oqtane Menu, and Login Components in the new Theme.

Setup external login and turn of AllowSiteLogin.

Result:

As you can see in the Video, the url of the login button does look correct but upon clicking i get taken to the page not found.

Interestingly, and a bit baffling, if i input the url directly into the browser, it works?

Here's the vid


https://github.com/user-attachments/assets/8aadee12-d23d-460d-a9b0-a18ff6d1b210

**Reply by Justincale** [reply]
Just to be clear. That is: copying the existing menu and login controls from Oqtane into the new theme.

**Comment by Justincale** [comment]
Any thoughts on the above?

cheers (and merry Xmas)

**Reply by sbwalker** [reply]
@Justincale I am not sure what you mean "copying the existing menu and login controls from Oqtane into the new theme"... usually for the login control you simply include a reference to the default component in your theme ie ```<Login />```

**Reply by Justincale** [reply]
Correct, but I require some slightly different functionality for my login control, so I place a copy of Oqtane’s login component in my theme and adapt it to my needs. When doing so, as in the above example, I receive the 404

**Reply by sbwalker** [reply]
@Justincale Since the standard Login component works fine then I would have to assume your customizations are causing the 404 issue you are experiencing. It is not possible to reproduce the problem or provide any guidance unless you share more details about the customizations.

**Comment by Justincale** [comment]
Hi Shaun,

Yes, i can confirm that after generating a new theme, copying the login control from Oqtane into the new theme, and attempting an oidc login using the new theme's login control that everything worked out, so there is definitely something going on with my mod's.

Thanks again for your time.
